src/components/QuranReader/index.tsx

```diff
@@ -55,47 +60,60 @@ const QuranReader = ({ initialData, chapter }: QuranReaderProps) => {
     },
     verseFetcher,
     {
-      initialData: initialData.verses,
+      initialData: isUsingDefaultTranslations ? initialData.verses : null, // initialData is set to null if the user changes/has changed the default translations so that we can prevent the UI from falling back to the default translations while fetching the verses with the translations the user had selected and we will show a loading indicator instead.
       revalidateOnFocus: false, // disable auto revalidation when window gets focused
       revalidateOnMount: true, // enable automatic revalidation when component is mounted. This is needed when the translations inside initialData don't match with the user preferences and would result in inconsistency either when we first load the QuranReader with pre-saved translations from the persistent store or when we change the translations' preferences after initial load.
     },
   );
   const readingView = useSelector(selectReadingView);
-  const pageLimit = isVerseView ? 1 : initialData.pagination.totalPages;
-  const verses = data.flat(1);
+  let body;
   let view;
-
-  if (readingView === ReadingView.QuranPage) {
-    view = <PageView verses={verses} />;
+  // if we are fetching the data (this will only happen when the user has changed the default translations so the initialData will be set to null).
+  if (!data) {
```

Comment:
I have mixed opinions here, thinking out loud. The nested if elses make the code a bit hard to follow. 


My first reaction is to remove the: 

```
 if (readingView === ReadingView.QuranPage) {
      view = <PageView verses={verses} />;
    } else {
      view = <TranslationView verses={verses} quranReaderStyles={quranReaderStyles} />;
    }

```

outside of the nested code block to flatten the structure. 


The other approach is to do something like this: 

```
if(!data) {
 return     
	<>
      <ContextMenu />
      <Container isSideBarVisible={isSideBarVisible}>
		<StyledLoading>loading...</StyledLoading>
	  </Container>
      <Notes />
    </>
}

// rest unchanged 
```

What do you think? No clear cut opinions

Reply:
I would go for the second approach but we would be duplicating the `ContextMenu`, `Container` and `Notes` in both cases which is not a problem for now but if we want to add more components inside `QuranReader` in the future, maybe we will have to take the common components into a separate component for example `QuranReaderContainer` or something like that like the previous PR I did and we can reuse it but for now I can do the second approach. I will include it in the upcoming commit.

Reply:
+1 to the second approach